cmake/projects/odb-compiler/hunter.cmake

```diff
@@ -29,9 +29,9 @@ elseif(CMAKE_HOST_UNIX)
       VERSION
       "2.4.0"
       URL
-      "http://www.codesynthesis.com/download/odb/2.4/odb-2.4.0-i686-linux-gnu.tar.bz2"
```

Comment:
> where is ARCHITECTURE_I686 set?

Nowhere, it's just first thing cames to mind. That's why I placed `???` near usage. Sorry for confusion.

I'm not 100% sure but you can use `CMAKE_SIZEOF_VOID_P`:
* http://stackoverflow.com/a/4029607/2288008